src/GafferScene/FilterResults.cpp

```diff
@@ -106,19 +106,24 @@ void FilterResults::affects( const Gaffer::Plug *input, AffectedPlugsContainer &
 {
 	ComputeNode::affects( input, outputs );
 
-	const ScenePlug *scenePlug = input->parent<ScenePlug>();
-	if( scenePlug && scenePlug == this->scenePlug() )
+	const ScenePlug *scenePlug = input->parent() == this->scenePlug() ? this->scenePlug() : nullptr;
```

Comment:
This looks correct, but something about how it's presented rubs me the wrong way - not a big deal, but something about a var named simply scenePlug that stores either the result of the scenePlug method or null depending on the input feels weird.  Maybe if it was named parentScenePlug to make it clear that it the scenePlug that is the parent of the input ( if it is a ScenePlug ) then it wouldn't have made me feel uneasy.

Reply:
I don't think `parentScenePlug` aids clarity particularly, because it also fails to communicate the "if and only if the parent is the same as `scenePlug()` part. I've just rewritten it without the variable in aad8965 - I think that's better?